### PR TITLE
#1526: Fixed missing import of the Cheerio type

### PR DESCRIPTION
**Pull request by sk-pub**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch.
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by lmiller1990** [comment]
Are you sure this is necessary? I am not seeing this typing error in `dev` in VSCode.

**Comment by sk-pub** [comment]
> Are you sure this is necessary? I am not seeing this typing error in `dev` in VSCode.

Yes. I cannot compile my project, which depends on @vue/server-test-utils, because of this.

And VS Code shows the error:
image

**Comment by lmiller1990** [comment]
I don't get this one. 

image

I will ask someone with more TS knowledge, please wait a moment.

**Comment by sk-pub** [comment]
Probably, you have Cheerio registered globally somehow.

**Comment by pikax** [comment]
@unicrus works fine for me, can you run `yarn --check-files`?

If that doesn't work, can you check if you have the file `node_modules/@types/cheerio/index.d.ts`?

EDIT: If still doesn't work, please remove `node_modules` folder and do a fresh install.

**Comment by sk-pub** [comment]
@lmiller1990, @pikax please take a look at the example to reproduce the bug: https://github.com/unicrus/vue-server-test-utils-1526.

TypeScript is a strictly typed language, all types must be defined before usage.

**Comment by pikax** [comment]
`@vue/server-test-utils` has `cheerio` and `@types/cheerio` has dependencies.

Your repo is overriding the `types` in tsconfig.json
```ts
  "types": [
      "@types/node"
    ]
```

Making typescript only load those types! As per the [documentation](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#types-typeroots-and-types)

```
If types is specified, only packages listed will be included
```


If you remove the types option, it will pick up the types by default. Or you can add Cheerio to your types
```
    "types": [
      "@types/node", 
      "@types/cheerio"
    ]
```

**Comment by sk-pub** [comment]
That's right. But @vue/test-utils does not rely on global types for everything else, like Vue or Vuex. They're explicitly imported. Why should Cheerio be different? It's not a well-known type.

**Comment by seyfer** [comment]
I cannot build my TS project because of this issue.

https://github.com/vuejs/vue-test-utils/issues/571#issuecomment-628749257

so I was forced to do this
`yarn remove @vue/server-test-utils @types/cheerio cheerio`

and rewrite my tests...

**Comment by lmiller1990** [comment]
@pikax any down sides to merge this types PR? I don't know TS super well, seems like it's blocking a few people - I don't see any downsides to the extra import (right?)

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "test:unit:karma:only": "cross-env TARGET=browser karma start test/setup/karma.conf.js --single-run",
     "test:unit:node": "yarn build:test && yarn test:unit:node:only",
     "test:unit:node:only": "cross-env TEST_ENV=node mocha-webpack --webpack-config test/setup/webpack.test.config.js test/specs/render.spec.js test/specs/renderToString.spec.js --require test/setup/mocha.setup.js",
-    "test:types": "tsc -p packages/test-utils/types && tsc -p packages/server-test-utils/types"
+    "test:types": "tsc -p packages/test-utils/types && tsc -p packages/server-test-utils/types --allowSyntheticDefaultImports"
```
Comment (pikax) [inline package.json]: You don't need to change this line

File: packages/server-test-utils/types/index.d.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import Vue, { VNodeData, ComponentOptions, FunctionalComponentOptions, Component } from 'vue'
 import { DefaultProps, PropsDefinition } from 'vue/types/options'
+import Cheerio from 'cheerio'
```
Comment (pikax) [inline packages/server-test-utils/types/index.d.ts]:
```suggestion
import * as Cheerio from 'cheerio'
```

**Review by pikax — CHANGES_REQUESTED** [review]

**Comment by pikax** [comment]
Sorry, got side tracked and this went under the radar.

My requested changes are not required, is just less code, if you want you can merge this.

The explanation is, cheerio types don't expose a default import, but when you import with typescript it expects to have the default export, the flag on the the script tells typescript to basically do `import * as ` if no default module is available.

**Comment by lmiller1990** [comment]
@unicrus if you can make those changes we can get this merged and released - we have a release in a day or two.

File: packages/server-test-utils/types/index.d.ts (hunk 2 above)
Comment (sk-pub) [reply packages/server-test-utils/types/index.d.ts]: Done

**Comment by sk-pub** [comment]
@lmiller1990, done. Thanks.

**Review by pikax — APPROVED** [review]

**Pull request merged**